pkg/vf/virtio.go

```diff
@@ -289,6 +292,112 @@ func (dev *VirtioVsock) AddToVirtualMachineConfig(vmConfig *VirtualMachineConfig
 	return nil
 }
 
+func (dev *NetworkBlockDevice) toVz() (vz.StorageDeviceConfiguration, error) {
+	if err := dev.validateNbdURI(dev.URI); err != nil {
+		return nil, fmt.Errorf("invalid NBD device 'uri': %s", err.Error())
+	}
+
+	if err := dev.validateNbdDeviceIdentifier(dev.DeviceIdentifier); err != nil {
+		return nil, fmt.Errorf("invalid NBD device 'deviceId': %s", err.Error())
+	}
+
+	attachment, err := vz.NewNetworkBlockDeviceStorageDeviceAttachment(dev.URI, dev.Timeout, dev.ReadOnly, dev.SyncronizationModeVZ())
+	if err != nil {
+		return nil, err
+	}
+
+	vzdev, err := vz.NewVirtioBlockDeviceConfiguration(attachment)
+	if err != nil {
+		return nil, err
+	}
+	err = vzdev.SetBlockDeviceIdentifier(dev.DeviceIdentifier)
+	if err != nil {
+		return nil, err
+	}
+
+	return vzdev, nil
+}
+
+func (dev *NetworkBlockDevice) validateNbdURI(uri string) error {
+	if uri == "" {
+		return fmt.Errorf("'uri' must be specified")
+	}
+
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return fmt.Errorf("error: %w", err)
+	}
+
+	// The format specified by https://github.com/NetworkBlockDevice/nbd/blob/master/doc/uri.md
+	if parsed.Scheme != "nbd" && parsed.Scheme != "nbds" && parsed.Scheme != "nbd+unix" && parsed.Scheme != "nbds+unix" {
```

Comment:
`NetworkBlockDeviceNew` in `pkg/config/virtio.go` says:
```
// The provided uri must be in the format nbd://<address>/<export-name>.
```
but we parse more than just `nbd://`. The https://github.com/NetworkBlockDevice/nbd/blob/master/doc/uri.md link would be useful to have in the doc for `NetworkBlockDeviceNew`

Reply:
Right. Updating 👍

Reply:
Updated the comment and the doc 👍